### ISSUE TITLE
Unexpected indent when pressing Enter at start of dedented line

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#3974]
## Environment data
-   Language Server version: 2023.2.31
-   OS and version: Windows 11 (10.0.25197)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.0

## Code Snippet

```python
def func():
    if True:
        x = 1
    print(x)
```

## Repro Steps
1. Put cursor before the `p` in `print`
2. Press Enter

## Expected behavior
New line is added, but indentation of `print(x)` remains unchanged.

## Actual behavior
`print(x)` is indented to be part of the `if` suite.

## Logs

```
[Info  - 5:44:22 PM] (29668) Pylance language server 2023.2.31 (pyright 085dfbfa) starting
...
[Info  - 5:44:22 PM] (29668) Auto-indent enabled
```

**Comment by fireattack** [comment]
Can confirm I encountered this bug. This is also a regression as the issue wasn't there a few days ago (assuming previous version). 

I use Py3.11 and Pylance 2023.2.30 (pyright dec6aa20) (is the version supposed to have a non-existing date?).

**Comment by netw0rkf10w** [comment]
As commented in https://github.com/microsoft/pylance-release/issues/3997, I tried downgrading VSCode to 1.74 as well as downgrading the Python and Pylance extensions to older versions from 4 months ago, but the problem still occurred.

Then, I tried changing the auto-indent mode from "Full" to "Keep", which didn't help.

Then, I tried changing the mode to "Brackets" and, it worked! I then tried re-installing the latest versions of VSCode and the extensions, and it was still working (with "Brackets").

Note that I made sure to restart VSCode at each of the above steps.

Then, a few hours later when I got home and resumed coding, it stopped working. Now I give up!

**Comment by debonte** [comment]
I don't think `editor.autoIndent` should have any impact on this behavior. In my experience, the editor's auto-indent always happens before ours. And then ours will overwrite whatever leading whitespace is present -- we don't add to it.

If it's really driving you crazy you could set the following until this is fixed:

```json
"[python]": {
  "editor.formatOnType": false
},
```

**Comment by netw0rkf10w** [comment]
@debonte Thanks a lot! That's exactly what I was looking for!
Maybe it was just a coincidence that it started to work for me (for a few hours) right after switching the auto-indent mode to "Brackets" (but then I wonder what was the true reason..., that's no longer important for me though).

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.2.43, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023243-24-February-2023-PreRelease)
